test/jdk/com/sun/jndi/ldap/objects/factory/LdapFactoriesFilterTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.net.URIBuilder;
+
+import javax.naming.NamingException;
+import javax.naming.Reference;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.SocketAddress;
+import java.util.Hashtable;
+
+/*
+ * @test
+ * @bug 8290368
+ * @summary Checks if LDAP specific objects factory filter system and security
+ *  properties can be used to restrict usage of object factories during
+ *  LDAP lookup operations.
+ * @modules java.naming/com.sun.jndi.ldap
+ * @library /test/lib ../../lib /javax/naming/module/src/test/test/
+ * @build LDAPServer LDAPTestUtils TestFactory
+ *
+ * @run main/othervm LdapFactoriesFilterTest false
```

Comment:
Are any of these filter property cases, malformed and would produce an error from ObjectInputFilter.create?

Reply:
There were no such cases.
Now, RMI and LDAP tests have been modified to run with malformed filter patterns which produce an error from `ObjectInputFilter.create`. 
Also, new cases were added to check that illegal patterns in one of RMI or LDAP filters not preventing other from being used.